### PR TITLE
[MM-45947] Added FAQ about using existing infrastructure

### PR DESCRIPTION
**Pull request by coltoneshaw**
#### Summary
The docs might be a bit verbose, just tried to give solid examples.

The existing database portion requires a workaround of setting the `InstanceEngine` to `aurora-mysql` to prevent a create command from running - https://github.com/mattermost/mattermost-load-test-ng/blob/2a982ad15971f9166f5c33d1d7263aa57baf7ad8/deployment/terraform/create.go#L387

I'm going to add another related PR to see if I can make that cancel early, same as when it encounters `aurora-mysql` in the string.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-45947.

### CODE REVIEW

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
This is really really good, thank you :pray: Just an unfinished sentence below, but otherwise is perfect.

**Review by isacikgoz — APPROVED** [review]
Looks good, nothing to add on top of what Alejandro requested.

File: docs/faq.md [hunk 1]
```diff
@@ -18,7 +18,69 @@ To manually run a bounded load-test using the [`coordinator`](https://github.com
...
+            "DriverName": "postgres",
+            "DataSource": "postgres://mmuser:mostest@databaseURL:port/mattermost?sslmode=disable\u0026connect_timeout=10\u0026binary_parameters=yes",
+            "DataSourceReplicas": [],
+            "DataSourceSearchReplicas": [],
+            "MaxIdleConns": 20,
+            "ConnMaxLifetimeMilliseconds": 3600000,
+            "ConnMaxIdleTimeMilliseconds": 300000,
+            "MaxOpenConns": 300,
+            "Trace": false,
+            "AtRestEncryptKey": "x85qyzyufe7aoh9a7upbyq6jrf5yas1r",
```
[the remark this answers is not in the document]
Comment (coltoneshaw) [reply docs/faq.md:66]: Maybe... No, it's just what was generated when setting up Mattermost for the first time. I can remove it.

File: docs/faq.md [hunk 2]
```diff
@@ -18,7 +18,69 @@ To manually run a bounded load-test using the [`coordinator`](https://github.com
...
 This type of load-test is used to determine the capacity of a system and will output an estimated number of users.
 The rule of thumb is that when starting an unbounded load-test we should always shoot for more users than what we think an installation can support.
-`ClusterConfig.MaxActiveUsers`  should be set to  `AgentInstanceCount * UsersConfiguration.MaxActiveUsers`.
+`ClusterConfig.MaxActiveUsers` should be set to `AgentInstanceCount * UsersConfiguration.MaxActiveUsers`.
+
+### Can I use a pre-existing Mattermost or database deployment?
+
+Yes, you can use an existing Mattermost deployment, or just the database portion.
+
+Note: You should **not** utilize an existing production setup to loadtest against because
```
[the remark this answers is not in the document]
Comment (coltoneshaw) [reply docs/faq.md:27]: So good, it doesn't even finish. Let me finish my thought here.

**Comment by coltoneshaw** [comment]
@agarciamontoro - As I'm walking a customer through these steps more we are hitting quite a few issues. I'm probably going to leave this pending until those are ironed out, just so we can properly document them.

TLDR on the issues:
They brought a database over to the loadtest that already had posts. These posts were missing files causing the loadtest to error early. The team had restrictions on domains, so it also failed there. Among a few other things that are stored directly in the database, not the config.

**Comment by agarciamontoro** [comment]
> These posts were missing files causing the loadtest to error early

If they have those files in an S3 bucket, then you can make use of this new config: https://github.com/mattermost/mattermost-load-test-ng/blob/master/docs/deployer_config.md#s3bucketdumpuri Although this is intended for public buckets. I know @sadohert has faced this problem before.

> The team had restrictions on domains, so it also failed there

What team and what domains?

 > Among a few other things that are stored directly in the database, not the config.
 
 Could you clarify this?
 
 I'm also ok discussing all this when you're finished, if you want to have a single discussion, and not one hundred of them :)

**Comment by coltoneshaw** [comment]
> Could you clarify this?

Channel and team settings are stored in the database. So, when bringing an existing database over to a loadtest you can have tons of tiny config that blocks the loadtest agents from accessing the team or channels.

> What team and what domains?

Basically the same as above. We had their main team domain restricted via settings, so no loadtest agents could join the team even though it was open.

> If they have those files in an S3 bucket

Let me have them try this. For now I had them do a new team and push the agents over there, since in the past Claudio has said it doesn't really matter _where_ the loadtest agents are, just that the database contains all the data. They originally wanted all the agents on their main team they use, but that's slowly posing problems.

**Comment by agarciamontoro** [comment]
No rush at all, but I'm curious of the current status of this.

**Comment by agarciamontoro** [comment]
Friendly ping, @coltoneshaw

**Comment by coltoneshaw** [comment]
@agarciamontoro - Let me get this done today. Sorry for the PR and ignoring. I've been juggling too many things!

**Comment by agarciamontoro** [comment]
There's no rush at all, I just like to keep PRs alive so I don't forget about them. No need to get this done today.

File: docs/faq.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (coltoneshaw) [reply docs/faq.md:66]: Okay, I added my thought here. I wasn't sure how to word it. But I've had three separate customers ask me how to setup their loadtest against prod, so i figured throwing a note that we could point to.

File: docs/faq.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (coltoneshaw) [reply docs/faq.md]: Okay, I added my thought here. I wasn't sure how to word it. But I've had three separate customers ask me how to setup their loadtest against prod, so i figured throwing a note that we could point to.

**Comment by coltoneshaw** [comment]
@agarciamontoro Okay, Updated it. Do you think it would be too much to add notes about the below? All issues I hit setting this up with a customer

- Enable email / password signin methods
- Turn on open server
- disable domain verification
- Ensure at least one team is set to open
- Increase the max users per team

File: docs/faq.md [hunk 3]
```diff
@@ -18,7 +18,69 @@ To manually run a bounded load-test using the [`coordinator`](https://github.com
...
 This type of load-test is used to determine the capacity of a system and will output an estimated number of users.
 The rule of thumb is that when starting an unbounded load-test we should always shoot for more users than what we think an installation can support.
-`ClusterConfig.MaxActiveUsers`  should be set to  `AgentInstanceCount * UsersConfiguration.MaxActiveUsers`.
+`ClusterConfig.MaxActiveUsers` should be set to `AgentInstanceCount * UsersConfiguration.MaxActiveUsers`.
+
+### Can I use a pre-existing Mattermost or database deployment?
+
+Yes, you can use an existing Mattermost deployment, or just the database portion.
+
+Note: You should **not** utilize an existing production setup to loadtest against because the loadtest agent will create users, posts, teams and channels to utilize and utilize most of your server resources. Best practice is to clone your production setup and loadtest against that.
```
Comment (agarciamontoro) [inline docs/faq.md:27]:
```suggestion
Note: You should **not** utilize an existing production setup to loadtest against because the loadtest agent will create users, posts, teams and channels and utilize most of your server resources. Best practice is to clone your production setup and loadtest against that.
```

**Review by agarciamontoro — CHANGES_REQUESTED** [review]
Just a typo below. Otherwise it looks great.

> @agarciamontoro Okay, Updated it. Do you think it would be too much to add notes about the below? All issues I hit setting this up with a customer
> 
>     * Enable email / password signin methods
> 
>     * Turn on open server
> 
>     * disable domain verification
> 
>     * Ensure at least one team is set to open
> 
>     * Increase the max users per team

I don't think it would be too much. Feel free to open a different PR with a proposal and we can review, but all documentation is welcome, more if it solves known issues!

**Comment by agarciamontoro** [comment]
I took the liberty to commit the typo fix. Will merge now :tada: Thank you, Colton!

**Review by agarciamontoro — APPROVED** [review]
